src/floating_exceptions/src/main.ts

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/gjs
+
+imports.gi.versions.Gtk = '3.0'
+
+const { Gio, GLib, Gtk, Pango } = imports.gi
+
+/** The directory that this script is executed from. */
+const SCRIPT_DIR = GLib.path_get_dirname(new Error().stack.split(':')[0].slice(1));
+
+/** Add our directory so we can import modules from it. */
+imports.searchPath.push(SCRIPT_DIR)
+
+import * as config from 'config';
+
+const { DEFAULT_RULES, Config } = imports.config;
+
+interface SelectWindow {
+    tag: 0
+}
+
+enum ViewNum {
+    MainView = 0,
+    Exceptions = 1
+}
+
+interface SwitchTo {
+    tag: 1
+    view: ViewNum
+}
+
+interface ToggleException {
+    tag: 2
+    wmclass: string | undefined
+    wmtitle: string | undefined
+    enable: boolean
+}
+
+interface RemoveException {
+    tag: 3
+    wmclass: string | undefined
+    wmtitle: string | undefined
+}
+
+type Event = SelectWindow
+    | SwitchTo
+    | ToggleException
+    | RemoveException
+
+interface View {
+    widget: any
+
+    callback: (event: Event) => void
+}
+
+function exceptions_button(): any {
+    let title = Gtk.Label.new("System Exceptions")
+    title.set_xalign(0)
+
+    let description = Gtk.Label.new("Updated based on validated user reports.")
+    description.set_xalign(0)
+    description.get_style_context().add_class("dim-label")
+
+    let icon = Gtk.Image.new_from_icon_name("go-next-symbolic", Gtk.IconSize.BUTTON)
+    icon.set_hexpand(true)
+    icon.set_halign(Gtk.Align.END)
+
+    let layout = Gtk.Grid.new()
+    layout.set_row_spacing(4)
+    layout.set_border_width(12)
+    layout.attach(title, 0, 0, 1, 1)
+    layout.attach(description, 0, 1, 1, 1)
+    layout.attach(icon, 1, 0, 1, 2)
+
+    let button = Gtk.Button.new()
+    button.relief = Gtk.ReliefStyle.NONE;
+    button.add(layout)
+
+    return button
+}
+
+class MainView implements View {
```

Comment:
@mmstick on typescript 4.x, if not prepending `export` the tsc will write `export {}` at bottom and gjs will complain it needs to be at the top. Float will not open